R/calculate_pga.R

```diff
@@ -130,23 +152,47 @@ calculate_pga = function(this_seg,
 
   # calculate total length of CNV
   affected_regions = this_seg %>%
-    dplyr::mutate(size = end - start) %>%
-    group_by(sample) %>%
-    summarise(total = sum(size))
+    dplyr::mutate(size = end - start)
+
+  if(per_chromosome){
+   affected_regions <- affected_regions %>%
+    group_by(sample, chrom) %>%
+    summarise(total = sum(size),
+              chrom_len = unique(chr_len))
+   affected_regions$PGA.chr = affected_regions$total/affected_regions$chrom_len
+     # add chr with 0
+   affected_regions = affected_regions %>%
+    dplyr::select(!all_of(c("total", "chrom_len"))) %>%
+    dplyr::mutate(PGA.chr = round(PGA.chr, 3)) %>%
+    pivot_wider(names_from = chrom, values_from = PGA.chr)
+
+      # now add any samples that can have 0 PGA
+   affected_regions = base::merge(sample_set,
+                                 affected_regions,
+                                 all.x = TRUE)
 
-  affected_regions$PGA = affected_regions$total / genome_size
+   affected_regions[is.na(affected_regions)] <- 0
 
-  # now add any samples that can have 0 PGA
-  affected_regions = base::merge(sample_set,
+   affected_regions = affected_regions %>%
+    rename("sample_id" = "sample")
+   colnames(affected_regions)[2:ncol(affected_regions)] <- paste0("chr",colnames(affected_regions)[2:ncol(affected_regions)],"_pga")
```

Comment:
I think this line needs to be conditional. For example if `projection = "hg38"` and `per_chromosome = TRUE` the returned object has double chr prefixed chromosomes.

```
seg <- get_sample_cn_segments(these_sample_ids = c("14-36022T",
                                                   "DOHH-2"))
seg <- dplyr::rename(seg, "sample" = "ID")

head(calculate_pga(this_seg = seg, 
                   projection = "hg38", 
                   per_chromosome = TRUE), 1)

  sample_id chrchr1_pga chrchr10_pga chrchr11_pga chrchr12_pga chrchr13_pga chrchr14_pga chrchr15_pga chrchr16_pga chrchr17_pga chrchr18_pga chrchr19_pga chrchr2_pga
1 14-36022T       0.913        0.977        0.967        0.966        0.835        0.811        0.801        0.858        0.917        0.909         0.79       0.962
  chrchr20_pga chrchr21_pga chrchr22_pga chrchr3_pga chrchr4_pga chrchr5_pga chrchr6_pga chrchr7_pga chrchr8_pga chrchr9_pga
1        0.607        0.703        0.631       0.945       0.979       0.965       0.652       0.974       0.637       0.785
```